examples/method/client_method_async_callback.cpp

```diff
@@ -0,0 +1,43 @@
+// This example requires the server example `server_method` to be running.
+
+#include <iostream>
+#include <thread>
+
+#include "open62541pp/open62541pp.h"
+
+int main() {
+    opcua::Client client;
+    client.connect("opc.tcp://localhost:4840");
+
+    // Browse method node
+    auto objNode = client.getObjectsNode();
+    auto greetMethodNode = objNode.browseChild({{1, "Greet"}});
+
+    // Run client event loop in separate thread
+    auto clientThread = std::thread([&] { client.run(0); });
+
+    // Set up a callback with a promise to determine when to exit the example.
+    std::promise<void> promise;
+    auto future = promise.get_future();
+    auto callback = [&promise](UA_StatusCode code, std::vector<opcua::Variant> output) {
```

Comment:
To distinguish this from the future approach, maybe just `std::cout` the results in the example?

Reply:
You mean std::cout in the callback function?
I have changed both examples to provide identical functionality, so I have changed the promise to also contain the output. Though I'm also fine with changing it back to the promise<void> and a printout in the callback function.

Reply:
Yes, just a callback with `std::cout` outputs results of the method call.
In my opinion, the manual `std::future` approach is very specific and shouldn't used in practice (because we have the other overload returning already a `std::future`).
We should try to make the examples as easy as possible, that's why I would prefer just a simple print in the callback.